server/routerlicious/packages/routerlicious-base/src/utils/sessionHelper.ts

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISession } from "@fluidframework/server-services-client";
+import { IDocument, ICollection } from "@fluidframework/server-services-core";
+import { getLumberBaseProperties, Lumberjack } from "@fluidframework/server-services-telemetry";
+
+/**
+ * Return to the caller with the status of the session.
+ */
+export async function getSession(ordererUrl: string,
+    historianUrl: string,
+    tenantId: string,
+    documentId: string,
+    documentsCollection: ICollection<IDocument>): Promise<ISession> {
+    const lumberjackProperties = getLumberBaseProperties(documentId, tenantId);
+
+    const tempDocument: IDocument = await documentsCollection.findOne({ documentId });
+    let tempSession: ISession = tempDocument.session;
+    if (tempSession === undefined) {
+        tempSession = {
+            ordererUrl,
+            historianUrl,
+            isSessionAlive: true,
+        };
+        await documentsCollection.upsert(
+            {
+                documentId,
+            },
+            {
+                deli: tempDocument.deli,
+                scribe: tempDocument.scribe,
+                session: tempSession,
+            },
+            {});
+        Lumberjack.info(`The Session ${JSON.stringify(tempSession)} was inserted into the document collection`,
+            lumberjackProperties);
+    }
+
+    let tempDeli = tempDocument.deli;
+    let tempScribe = tempDocument.scribe;
+    const isSessionAlive: boolean = tempSession !== undefined ? tempSession.isSessionAlive : true;
+    if (tempSession !== undefined && !tempSession.isSessionAlive) {
+        // Reset logOffset, ordererUrl, and historianUrl when switching cluster.
+        if ((tempSession.ordererUrl !== null && tempSession.ordererUrl !== ordererUrl) ||
```

Comment:
Curious, why `null` instead of `undefined`?

Reply:
Got it. It should be undefined.